### ISSUE TITLE
How to show review stars within collection templates?

### ISSUE THREAD

**Issue by atomgiant** [Shopify/product-reviews-sample-app#1]
I'd like to start a discussion about how to show the review stars for this sample app inside of a collection or search template. 

For example, for each product in the collection, I'd like to see the average rating stars when viewing a collection or search template.

This use case doesn't seem to be directly supported with the new App Blocks unless I am mistaken. Do you have any tips you can share about how you might make this work? 

Thanks,
Tom

**Comment by govindrai** [comment]
Wondering the same.

**Comment by atomgiant** [comment]
I don't believe OS2.0 supports this use case so there is no simple option that is OS2.0 native.

I decided to wait until after the holidays to decide which path to take to add app blocks to collection items. 

I believe there are really only 2 options for this:

* Manual snippet - use a standard snippet within the theme the "old-fashioned" way
* App Embed block - use an App Embed block with JavaScript to insert the content dynamically. The content would need to be added to the page in a hidden way and then inserted for each product

The App Embed block would be a very brittle process and need to change for each theme. So based on this, I'm leaning towards just using the Manual snippet option for collection items, and App blocks for product pages.

IMO this is a glaring flaw with OS2.0 so I hope they'll add some form of support in the future.

**Comment by atomgiant** [comment]
There are still no good options. If you want to use Liquid you have no choice but to modify the theme. I'm moving to web components and the Storefront API to load the data. There is the new theme blocks just announced that in theory allow nested blocks. However, I haven't looked to see if they intend to support collection grid item theme blocks. I've given up on Liquid being a viable choice for theme apps.
